add percentage (DPT_Scaling) KNX sensors

tiktok7 opened the pull request:
1. moved basic functionality to KNXSensorBaseClass instead of
KNXSensorFloatClass
2. all derived classes just override the "convert" method to handle different data formats
3. added "if" clause in setup for a "percentage" sensor type and added KNXSensorDPTScalingClass with DPT_Scaling conversion (raw: 0 - 255 = display: 0 - 100)

## Description:
This change is intended to provide support for additional data items on devices such as thermostats/covers that use the scaled percentage data format (1 byte). The existing sensors use the 2 byte float data format.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2876

## Example entry for `configuration.yaml` (if applicable):
```yaml
 - platform: knx
   name: percent
   type: percentage
   address: 1/0/5

```

## Checklist:
__Note__: there are no new dependencies and no new files

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.


[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

tiktok7 committed:
add percentage (DPT_Scaling) KNX sensors

1. moved basic functionality to KNXSensorBaseClass instead of
KNXSensorFloatClass
2. added "if" clause in setup for a "percentage" sensor type and added KNXSensorDPTScalingClass

tiktok7 committed:
support-knx-percentage-sensor: lint correction

Updated convert method base sensor class to avoid lint warning
(R201 - Method could be a function)

open-homeautomation commented:
👍

balloob reviewed (changes requested):
Please add a `PLATFORM_SCHEMA` based on voluptuous that validates `CONF_TYPE` is one of the approved values.

Example from KNX light: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/light/knx.py#L20-L24

tiktok7 commented:
No problem. Thanks for the quick feedback.

tiktok7 committed:
added PLATFORM_SCHEMA for configuration

1. added SCHEMA extension for defined keywords
2. moved fixed data for internal settings out of sensor logic
3. moved everything into standard KNXSensor object
4. added parsing of extra config parameters in __init__

tiktok7 committed:
correct lint errors on support-knx-percentage-sensor

tiktok7 commented:
@balloob : that's the best I could do

balloob commented:
Thanks!

tiktok7 commented:
@daBONDi @Julius2342 @open-homeautomation : I should have thought of asking earlier but you three seem to have made the biggest changes on the knx platform to date. I had to make a larger than expected change to this PR to add in a PLATFORM_SCHEMA so it may have side-effects on knx sensors but don't have enough devices to test all types. Do you think you could check your devices for any issues for the lux/speed/temp sensors on the latest dev branch?

open-homeautomation commented:
Temp sensors here are working fine. I don't have any Lux/speed sensors, but I would not expect problems.
